numba_cuda/numba/cuda/tests/cudadrv/test_linker.py

```diff
@@ -329,6 +329,20 @@ def test_get_local_mem_per_specialized(self):
         calc_size = np.dtype(np.float64).itemsize * LMEM_SIZE
         self.assertGreaterEqual(local_mem_size, calc_size)
 
+    def test_link_for_different_cc(self):
+        if not config.CUDA_USE_NVIDIA_BINDING:
+            self.skipTest("Ctypes Linker does not support outputting PTX.")
+
+        linker = _Linker.new(cc=(12, 0))
```

Comment:
The error only occurs when the LTOIR CC is greater than the Linker's given CC, so this will never fail. We also need to turn LTO on to trigger the issue. On my system (CC 8.6), I can use

```suggestion
        linker = _Linker.new(cc=(7, 5), lto=True)
```

to trigger it.

Reply:
Right - the cc argument is passed down to the linker, and this PR is to make sure the nvrtc compile is of the same version as the linker argument.